### ISSUE TITLE
Principle #12 "Naming conventions"

### ISSUE THREAD

**Issue by nataled** [OBOFoundry/OBOFoundry.github.io#958]
Current wording:

NOTE
-------

The content of this page is scheduled to be reviewed. Improved wording will be posted as it becomes available.

Details
-------

For full details, see this paper: <http://www.biomedcentral.com/1471-2105/10/125>

Briefly, some important things to remember:

 * use rdfs:label for the primary label
 * include exactly one rdfs:label for every declared entity (e.g. class, property)
 * write labels, synonyms, etc as if writing in plain English text. ie use spaces to separate words, only capitalize proper names (e.g. Parkinson disease). Do not use CamelCase, do_not_use_underscores
 * spell out abbreviations. Abbreviations can be included as a separate property.
 * make the primary labels to be as unambiguous as possible. Remember, your ontology may be used in a different context than that for which it was originally intended. Remember also of course that the label should be unambiguous without looking at parent terms
 * labels should be unique within an ontology
 * use the IAO property 'obo foundry unique label' [http://purl.obolibrary.org/obo/IAO_0000589](http://purl.obolibrary.org/obo/IAO_0000589) to declare a pan-OBO unique label if required

**Comment by nataled** [comment]
Wording suggestions and discussion:

Also see http://msi-ontology.sourceforge.net/namingconventions/Naming_Conv_v14.htm


Sez Melissa: Naming is complicated as per the referenced documentation. Do you think it is possible to provide a high level summary here so that we can use this for the purposes of review? Basically a short list of requirements and consistency checks.
We agreed.

Name: Naming conventions

Summary and Purpose: the names for classes in an ontology should be intelligible to scientists and be amenable to natural language processing.

Implementation:



[todo: put MUST and SHOULD in caps.]

Here is a list of the headline points from the above links.

Names of classes should as far as possible reflect usage in ordinary scientific body text. They are not titles. This means that
words within them should only be capitalized if they are proper names (for example Golgi)
multiword names should have the words separated by spaces (not underscores or CamelCase)
non-deprecated class names should not contain information relevant only to the ontology, for example “protocol (definition incomplete)”. Reasons for allowing this for deprecated terms include to avoid clashes between classes which would otherwise have the same name, and to provide a readily-usable audit trail.
Short, commonly-used names should appear as they do in ordinary running text in the normal order of words and should not contain commas, hence “hoses for the use of firemen”, not “hoses, firemen, for the use of”. Commas may be used where they are part of, say, chemical names, or in long complicated terms (more than four or five words) such as in the Gene Ontology.
names should be singular: “plant cell” rather than “plant cells”. If you are referring to a collective or aggregate, then say so. Plurals may be added as synonyms.

Names of classes should be short and memorable, for example “wall of esophagus” rather than “the wall of the esophagus”.

Non-obvious abbreviations should be spelt out, for example “nuclear magnetic resonance” rather than “NMR”. DNA and laser are clear exceptions to this. Abbreviations should be added as synonyms.

MERGE WITH PROPOSED ‘UNIQUE LABEL’ PRINCIPLE: Names should be univocal within a given ontology, for example “nuclear magnetic resonance magnet” rather than “magnet”. 

Discussion point: what about names of relations?

**Comment by cmungall** [comment]
I do not think that linked MSI doc is useful. Way too much detail. On scanning it I see stuff that is not irrelevant that I disagree with. E.g. out of date recommendations to avoid multiple inheritance. [my opinions on this](https://douroucouli.wordpress.com/2019/05/10/ontotip-single-inheritance-principle-considered-dangerous/).

**Comment by cmungall** [comment]
"be amenable to natural language processing" --> don't necessarily agree with this. I think they key point is that it should reflect normal scientific terminology. NLP should follow naturally from that.

**Comment by cmungall** [comment]
I think the text in the 2nd comment is largely redundant. The existing wording is quite clear and succinct IMHO. I see a few useful additional suggestions in the text, that can be added as new bullets:

 * do not use "the" in primary labels
 * use normal word ordering
 * do not include metadata in the label

**Comment by nataled** [comment]
Add bullet point stating there should be no additional leading or trailing whitespace see#257

**Comment by nlharris** [comment]
what's the status of this principle?
